Packs/rasterize/Integrations/rasterize/rasterize.py

```diff
@@ -77,449 +66,452 @@ class RasterizeType(Enum):
+    exc_value = args.exc_value
+    if args.exc_type == json.decoder.JSONDecodeError:
+        demisto.debug(f"Caught a JSONDecodeError exception in _recv_loop, suppressing, {exc_value}")
+    else:
+        demisto.info(f"Unsuppressed Exception in _recv_loop: {args.exc_type=}, {exc_value=}")
 
 
-def return_err_or_warn(msg):
-    return_error(msg) if WITH_ERRORS else return_warning(msg, exit=True)
+class TabLifecycleManager:
+    def __init__(self, browser, chrome_port, offline_mode):
+        self.browser = browser
+        self.chrome_port = chrome_port
+        self.offline_mode = offline_mode
+        self.tab = None
 
+        demisto.debug(f'TabLifecycleManager, __init__, {self.chrome_port=}, active tabs len: {len(self.browser.list_tab())}')
 
-def opt_name(opt):
-    return opt.split('=', 1)[0]
+    def __enter__(self):
+        self.tab = self.browser.new_tab()
 
+        self.tab.start()
+        if self.offline_mode:
+            self.tab.Network.emulateNetworkConditions(offline=True, latency=-1, downloadThroughput=-1, uploadThroughput=-1)
+        else:
+            self.tab.Network.emulateNetworkConditions(offline=False, latency=-1, downloadThroughput=-1, uploadThroughput=-1)
 
-def merge_options(default_options, user_options):
-    """merge the defualt options and user options
+        self.tab.Page.enable()
+        demisto.debug(f'TabLifecycleManager, {self.chrome_port=}, entering tab {self.tab.id},'
+                      f' tabs len: {len(self.browser.list_tab())}')
+        return self.tab
 
-    Arguments:
-        default_options {list} -- list of options to use
-        user_options {string} -- user configured options comma seperated (comma value can be escaped with \\)
+    def __exit__(self, exc_type, exc_val, exc_tb):  # pylint: disable=unused-argument
+        if exc_type or exc_val or exc_tb:
+            demisto.info(f'TabLifecycleManager, __exit__ with exception, {self.chrome_port=}, {exc_type=}, {exc_val=}, {exc_tb=}')
 
-    Returns:
-        list -- merged options
-    """
-    user_options = re.split(r'(?<!\\),', user_options) if user_options else []
-    if not user_options:  # nothing to do
-        return default_options.copy()
-    demisto.debug(f'user chrome options: {user_options}')
-    options = []
-    remove_opts = []
-    for opt in user_options:
-        opt = opt.strip()
-        if opt.startswith('[') and opt.endswith(']'):
-            remove_opts.append(opt[1:-1])
-        else:
-            options.append(opt.replace(r'\,', ','))
-    # remove values (such as in user-agent)
-    option_names = [opt_name(x) for x in options]
-    # add filtered defaults only if not in removed and we don't have it already
-    options.extend([x for x in default_options if (opt_name(x) not in remove_opts and opt_name(x) not in option_names)])
-    return options
+        if self.tab and self.tab.id:
+            tab_id = self.tab.id
+            threading.excepthook = excepthook_recv_loop
```

Comment:
Please add a comment explaining why you're suppressing this specific exception (it isn't evident from the code)

Reply:
Done.